src/components/SelectField/__tests__/SelectField.spec.tsx

```diff
@@ -0,0 +1,283 @@
+import React from 'react';
+import {
+  expect,
+  test,
+} from '@playwright/experimental-ct-react';
+import {
+  mixPropTests,
+  propTests,
+} from '../../../../tests/playwright';
+import {
+  SelectFieldForTest,
+  SelectFieldForRefTest,
+  SelectFieldForFormLayoutTests,
+} from './SelectField.story';
+import type { SelectFieldForFormLayoutTestsProps } from './SelectField.story';
+import { openSelectFieldOptionsTest } from './_propTests/openSelectFieldOptionsTest';
+
+const groupedOptions = [
+  {
+    key: 'optgroup_key',
+    label: 'optgroup',
+    options: [
+      {
+        disabled: false,
+        label: 'option1',
+        value: 'value1',
+      },
+      {
+        disabled: true,
+        label: 'option2',
+        value: 'value2',
+      },
+    ],
+  },
+];
+
+const baseOptions = [
+  {
+    disabled: false,
+    key: 'key1',
+    label: 'option1',
+    value: 'value1',
+  },
+  {
+    disabled: false,
+    label: 'option2',
+    value: 'value2',
+  },
+];
+
+test.describe('SelectField', () => {
+  test.describe('base', () => {
+    test.describe('visual', () => {
+      [
+        ...propTests.defaultComponentPropTest,
+        ...propTests.labelPropTest,
+        ...propTests.renderAsRequiredPropTest,
+        ...propTests.isLabelVisiblePropTest,
+        ...propTests.requiredPropTest,
+        ...propTests.sizePropTest,
+        ...mixPropTests([
+          propTests.layoutPropTest,
+          propTests.fullWidthPropTest,
+        ]),
+        ...mixPropTests([
+          propTests.disabledPropTest,
+          propTests.variantPropTest,
+          propTests.validationStatePropTest,
+        ]),
+        ...mixPropTests([
+          propTests.validationTextPropTest,
+          propTests.validationStatePropTest,
+        ]),
+        ...mixPropTests([
+          propTests.validationTextPropTest,
+          propTests.helpTextPropTest,
+        ]),
+      ].forEach(({
+        name,
+        onBeforeTest,
+        onBeforeSnapshot,
+        props,
+      }) => {
+        test(name, async ({
+          mount,
+          page,
+        }) => {
+          if (onBeforeTest) {
+            await onBeforeTest(page);
+          }
+
+          const component = await mount(
+            <SelectFieldForTest
+              {...props}
+            />,
+          );
+
+          if (onBeforeSnapshot) {
+            await onBeforeSnapshot(page, component);
+          }
+
+          const screenshot = await component.screenshot();
```

Comment:
To fixed errors caused by animations, pass following options into `screenshot()`:

```
{
          animations: 'disabled',
}
```

This should be used for all screenshot calls of this component.

Reply:
See: https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-screenshot-1-option-animations